magda-web-client/src/Components/Common/DataPreviewMapOpenInNationalMapButton.tsx

```diff
@@ -67,7 +68,52 @@ class DataPreviewMapOpenInNationalMapButton extends Component<PropsType> {
                 ]
             };
         } else {
+            const dataUrl = distribution?.downloadURL
+                ? distribution.downloadURL
+                : distribution.accessURL;
+
+            let layers: any = undefined;
             const id = "data.gov.au-postMessage-" + distribution?.identifier;
+            if (dataUrl) {
+                const queries = URI.parseQuery(dataUrl);
+                layers = queries["layers"];
+                if (typeof layers !== "string") {
+                    layers = queries["LAYERS"];
+                }
+
+                if (typeof layers !== "string") {
+                    layers = queries["Layers"];
+                }
+            }
+
+            const type =
+                distribution?.format.toLowerCase() === "wms"
```

Comment:
```suggestion
                distribution?.format?.toLowerCase() === "wms"
```
we used optional chaining for `format` which is good
But when distribution?.format eval to undefined,  it will throw an error when call `toLowerCase()` on undefined.
you can use `distribution?.format?.toLowerCase()` instead --- optional chaining on method call will eval to `undefined`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls

Reply:
done